contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3715,3 +3660,100 @@ IN query text)
 RETURNS SETOF RECORD
 AS 'babelfishpg_tsql', 'openquery_internal'
 LANGUAGE C VOLATILE;
+
+CREATE OR REPLACE FUNCTION sys.fn_listextendedproperty
+(
+    "@name" sys.sysname,
+    "@level0type" VARCHAR(128),
+    "@level0name" sys.sysname,
+    "@level1type" VARCHAR(128),
+    "@level1name" sys.sysname,
+    "@level2type" VARCHAR(128),
+    "@level2name" sys.sysname
+)
+RETURNS TABLE (
+    objtype sys.sysname,
+    objname sys.sysname,
+    name sys.sysname,
+    value sys.sql_variant
+)
+AS $$
+DECLARE
+    var_dbid SMALLINT;
+    var_schema_name NAME;
+    var_major_name NAME;
+    var_minor_name NAME;
+    var_type TEXT;
+    rec record;
+    objtype sys.sysname;
+    objname sys.sysname;
+BEGIN
+    "@level0type" := UPPER("@level0type");
+    "@level1type" := UPPER("@level1type");
+    "@level2type" := UPPER("@level2type");
+
+    var_dbid := sys.db_id();
+    var_schema_name := '';
+    var_major_name := '';
+    var_minor_name := '';
+    var_type := '';
+
+    -- DATABASE
+    IF "@level0type" IS NULL THEN
+        var_type := 'DATABASE';
+    END IF;
+
+    -- SCHEMA or object in SCHEMA
+    IF "@level0type" IN ('SCHEMA') THEN
+        var_schema_name := LOWER("@level0name");
```

Comment:
I can confirm that fn_listextendedproperty(), as is common in SQL Server, ignores trailing spaces for its arguments, but not leading spaces (i.e. when leading spaces are present, these are not stripped, and that usually means the object is not found)

Reply:
In addition ,please note that the examples for fn_listextendedproperty() use **default** as argument a lot, which Babelfish currently does not support. That is not an issue for this PR, but a generic issue that would have to be solved in a different context. However, this may mean the usefulness of fn_listextendedproperty() may be limited as a result